File: app/src/main/java/dev/expositopablo/tonedeath/androidutils/ImagesUtils.java
package dev.expositopablo.tonedeath.androidutils;

import android.graphics.Color;
import android.graphics.PorterDuff;
import android.graphics.drawable.Drawable;

public class ImagesUtils {

    //https://stackoverflow.com/a/14128907/5946253
    public static Drawable convertDrawableToGrayScale(Drawable drawable) {
        if (drawable == null) {
            return null;
        }
        Drawable res = drawable.mutate();
        res.setColorFilter(Color.GRAY, PorterDuff.Mode.SRC_IN);
        return res;
    }
}

File: app/src/main/java/dev/expositopablo/tonedeath/androidutils/ImagesUtils.kt
package dev.expositopablo.tonedeath.androidutils

import android.graphics.Color
import android.graphics.PorterDuff
import android.graphics.drawable.Drawable

object ImagesUtils {

    //https://stackoverflow.com/a/14128907/5946253
    fun convertDrawableToGrayScale(drawable: Drawable?): Drawable? {
        if (drawable == null) {
            return null
        }
        val res = drawable.mutate()
        res.setColorFilter(Color.GRAY, PorterDuff.Mode.SRC_IN)
        return res
    }
}
